### PR TITLE
Settings; customize app icon and theme override

### PR DESCRIPTION
**Pull request by jedireza**
This PR adds the ability to change the app icon and also change the theme of the app separate from the system theme.

### How was this tested?
- [x] I tested this manually

### Manual test cases
I tested in the simulator.

#### Change app icon

https://user-images.githubusercontent.com/979929/169635672-851ce4ce-2653-4b04-bacd-59dc90b8d49e.mp4

#### Change theme

https://user-images.githubusercontent.com/979929/169635748-13100cbf-bf37-46bc-a0c2-07478c6e2218.mp4

#### Change theme (multiple instances)

https://user-images.githubusercontent.com/979929/169664280-24c1ac0f-5dd8-4997-9259-df94a8c9ad68.mp4

## Issues addressed
N/A

### CODE REVIEW

File: Client/Application/SceneDelegate.swift [hunk 1]
```diff
@@ -13,6 +13,7 @@ import StoreKit
 private let log = Logger.browser
 
 class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+    private(set) static var shared: SceneDelegate?
```
Comment (evancrow) [inline Client/Application/SceneDelegate.swift:16]: Strongly against a static `SceneDelegate` instance. I recommend using one of the `getCurrentSceneDelegate` methods, and if you can't plumb through any of the parameters, as a last resort use `getCurrentSceneDelegateOrNil`.

File: Client/Frontend/Settings/CustomizeSettingsSection.swift [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    Picker(selection: $selectedTheme, label: EmptyView()) {
+                        ForEach(CustomizeThemeOption.allCases, id: \.self) { option in
+                            Text(option.localizedName)
+                                .tag(option)
+                        }
+                    }
+                    .pickerStyle(.inline)
+                }
+                .onChange(of: selectedTheme) { option in
+                    Defaults[.customizeTheme] = option
```
Comment (evancrow) [inline Client/Frontend/Settings/CustomizeSettingsSection.swift:33]: Should probably move a lot of this to the `SceneDelegate` or a separate model so we don't have lots of logic inside a `View`.

File: Client/Application/SceneDelegate.swift [hunk 3]
```diff
@@ -26,12 +27,23 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
...
+
         self.scene = scene
         guard let scene = (scene as? UIWindowScene) else { return }
 
         window = .init(windowScene: scene)
         window!.makeKeyAndVisible()
 
+        // cutomized theme preference
+        if Defaults[.customizeTheme] != .system {
+            if Defaults[.customizeTheme] == .dark {
```
Comment (evancrow) [inline Client/Application/SceneDelegate.swift:40]: Likely can just write this as a `ternary`, not too important though

File: Client/Frontend/Settings/CustomizeSettingsSection.swift [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            HStack {
+                Text("Theme")
+                Spacer()
+                Text(selectedTheme.rawValue)
+                    .foregroundColor(.secondary)
+            }
+        }
+    }
+
+    @ViewBuilder var iconNavLink: some View {
```
Comment (evancrow) [inline Client/Frontend/Settings/CustomizeSettingsSection.swift:62]: nit: start `var` on a new line underneath `@ViewBuilder`

**Review by evancrow — CHANGES_REQUESTED** [review]
Super neat, just a few things I think should be addressed.

File: Client/Application/SceneDelegate.swift (hunk 1 above)
Comment (darinwf) [reply Client/Application/SceneDelegate.swift:16]: Yeah, on iPad the user can create multiple browser windows and each one is represented by a scene. We end up with N `SceneDelegate` instances in other words. Each has its own `BVC` and `TabManager`.

**Review by darinwf — DISMISSED** [review]

**Comment by darinwf** [comment]
Let's review the UI changes in iOS design sync on Monday.

File: Client/Application/SceneDelegate.swift (hunk 1 above)
Comment (jedireza) [reply Client/Application/SceneDelegate.swift]: Thanks for bringing this up! I adjusted the cost to handle all scenes and uploaded a new video demonstrating this working.

File: Client/Frontend/Settings/CustomizeSettingsSection.swift (hunk 2 above)
Comment (jedireza) [reply Client/Frontend/Settings/CustomizeSettingsSection.swift:33]: Yep, done.

File: Client/Application/SceneDelegate.swift (hunk 3 above)
Comment (jedireza) [reply Client/Application/SceneDelegate.swift]: This is gone now.

File: Client/Frontend/Settings/CustomizeSettingsSection.swift (hunk 4 above)
Comment (jedireza) [reply Client/Frontend/Settings/CustomizeSettingsSection.swift:54]: Maybe we should consider the `wrapAttributes` rule for our swift format options? https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#wrapattributes

**Review by evancrow — APPROVED** [review]
Code LGTM, thanks! should still wait for design approval :)

**Comment by Ria-yu** [comment]
Designer input:

We sync-ed on this in iOS Design Open Hours [(notes)](https://paper.dropbox.com/doc/Mobile-Design-Open-Hours-iOS-Android--BiNeTHC~wGLgqDKWc4zSfSYLAg-94JrusTJaq0a3NRhIgcqk#:uid=816781856239286097298723&h2=May-25%2C-2022---iOS) and agreed on these decisions: 
        - Section name: ‘Appearance’
        - Theme choice: Ship as-is. Decouple web / app syncing, and investigate all possible cases with and without Neeva browser context, mobile and desktop.
        - Icon choices:  
            - ship with only 2 options (light Neeva logo, and dark Neeva logo)
            - Dark Neeva logo: 
                - There’s no official logo today. 
                [x] AI:  Ria to make an official dark logo v1 — Thanks Ria! [@Ria Y]
                - Logo assets Figma: https://www.figma.com/file/pQX1oZi3WFeIVTazdOKhST/Logos?node-id=2707%3A865

According to Reza's updates in #design-mobile Slack channel, the decisions above are reflected in the latest PR. In that case, designers are happy to approve this :).

**Pull request merged**